core/domain/voiceover_services.py

```diff
@@ -0,0 +1,291 @@
+    voiceover_application_model = _get_voiceover_application_model(
+        voiceover_application)
+
+    voiceover_application_model.put()
+
+
+def get_voiceover_application_by_id(voiceover_application_id):
+    """Returns voiceover application model corresponding to give id.
+
+    Args:
+        voiceover_application_id: str. The voiceover application id.
+
+    Returns:
+        BaseVoiceoverApplication. The voiceover application object for the give
+        application id.
+    """
+    voiceover_application_model = (
+        suggestion_models.GeneralVoiceoverApplicationModel.get_by_id(
+            voiceover_application_id))
+    return _get_voiceover_application_from_model(voiceover_application_model)
+
+
+def get_reviewable_voiceover_applications(user_id):
+    """Returns a list of voiceover applications which the given user can review.
+
+    Args:
+        user_id: str. The user ID of the reviewer.
+
+    Returns:
+        list(BaseVoiceoverApplication). A list of voiceover application which
+        the given user can review.
+    """
+    voiceover_application_models = (
+        suggestion_models.GeneralVoiceoverApplicationModel
+        .get_reviewable_voiceover_applications(user_id))
+
+    return [
+        _get_voiceover_application_from_model(model) for model in (
+            voiceover_application_models)]
+
+
+def get_user_submitted_voiceover_applications(user_id, status=None):
+    """Returns a list of voiceover application submitted by the given user which
+    are currently in the given status.
+
+    Args:
+        user_id: str. The id of the user.
+        status: str|None. The status of the voiceover application.
+
+    Returns:
+        BaseVoiceoverApplication). A list of voiceover application which are
+        submitted by the given user.
+    """
+    voiceover_application_models = (
+        suggestion_models.GeneralVoiceoverApplicationModel
+        .get_user_voiceover_applications(user_id, status))
+
+    return [
+        _get_voiceover_application_from_model(model) for model in (
+            voiceover_application_models)]
+
+
+def accept_voiceover_application(voiceover_application_id, reviewer_id):
+    """Accept the voiceover application of given voiceover application id.
+
+    Args:
+        voiceover_application_id: str. The is of the voiceover application which
+            need to be accepted.
+        reviewer_id: str. The user ID of the reviewer.
+    """
+    voiceover_application = get_voiceover_application_by_id(
+        voiceover_application_id)
+    if reviewer_id == voiceover_application.author_id:
+        raise Exception(
+            'Applicants are not allowed to review their own '
+            'voiceover application.')
+
+    reviewer = user_services.UserActionsInfo(user_id=reviewer_id)
+
+    voiceover_application.accept(reviewer_id)
+
+    _save_voiceover_application(voiceover_application)
+
+
+    if voiceover_application.target_type == feconf.ENTITY_TYPE_EXPLORATION:
```

Comment:
If the type is not this, throw an exception

Reply:
Should we throw an exception after saving the application model on line 192?

Reply:
I'm not sure how I should make sure that all the three (updating application model, assigning role and sending email) happens correctly if any of them fails then revert all the changes. Do you have any suggestions on whether we can achieve this (or do we want this)?

Reply:
Actually what I meant is if voiceover_application.target_type is not feconf. ENTITY_TYPE_EXPLORATION, then an exception must be thrown.

Reply:
> I'm not sure how I should make sure that all the three (updating application model, assigning role and sending email) happens correctly if any of them fails then revert all the changes. Do you have any suggestions on whether we can achieve this (or do we want this)?

I don't think this can be possible. By validating the models before we do all the save operations, I think we can assume that this will happen with very low probability.

Reply:
> Actually what I meant is if voiceover_application.target_type is not feconf. ENTITY_TYPE_EXPLORATION, then an exception must be thrown.

Then it won't be possible to write the test as the exception won't be reachable. Line 181 and 192 will validate the value of the target_type, I've added the IF over here so that in future if we are going to have different target_type (with its class) then we can add the else case, I can remove the `if` check if you want?

Reply:
Ah I see. No, keep the if check I think.